app/src/main/java/com/codesoom/assignment/controller/ProductController.java

```diff
@@ -0,0 +1,65 @@
+// TODO: Cotroller List 구현하기 (테스트 코드 통과)
+// TODO: Cotroller Detail 구현하기
+// TODO: Cotroller Create 구현하기
+// TODO: Cotroller update 구현하기
+// TODO: Cotroller patch 구현하기
+// TODO: Cotroller delete 구현하기
+
+package com.codesoom.assignment.controller;
+
+import com.codesoom.assignment.ProductNotFoundException;
+import com.codesoom.assignment.domain.Product;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/products")
+@CrossOrigin
+public class ProductController {
+    Product product = new Product();
+
+    private List<Product> products = new ArrayList<>();
+    private Long newId = 0L;
+
+    public ProductController(Product product) {
+        this.product = product;
+    }
+
+    @GetMapping
+    public List<Product> list() {
+        return products;
+    }
+
+    @GetMapping("{id}")
+    public Product detail(Long id) {
+        return products.stream()
+                .filter(product -> product.getId().equals(id))
+                .findFirst()
+                .orElseThrow(() -> new ProductNotFoundException(id));
```

Comment:
어떤 예외 상황인지 분명하게 알려줘야 디버깅하는 사람이 야근을 덜 합니다.
```suggestion
                .orElseThrow(() -> new ProductNotFoundException(String.format("상품 아이디 %d 에 해당하는 상품을 찾지 못했습니다.", id)));
```
여기에서 `String.format`을 사용한 이유는 `id`가 `null`인 경우를 방지하기 위해서입니다.